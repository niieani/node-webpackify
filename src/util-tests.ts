import {resolve, join} from 'path'
import execa from 'execa'

export function runFixture(name: string) {
  const fixturePath = resolve(__dirname, '..', 'fixtures', name)
  return execa(
    'node',
    ['-r', join(fixturePath, 'register'), fixturePath]
  )
}
