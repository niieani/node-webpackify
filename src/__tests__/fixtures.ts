import {runFixture} from '../util-tests'

describe('uses a loader', () => {
  jest.setTimeout(10000)

  it ('raw-loader', async () => {
    const result = await runFixture('raw')
    expect(result.failed).toBe(false)
    expect(result.stdout).toMatchSnapshot()
  })

  it ('css-loader and style-loader', async () => {
    const result = await runFixture('css')
    expect(result.failed).toBe(false)
    expect(result.stdout).toMatchSnapshot()
  })

  it ('postcss-loader', async () => {
    const result = await runFixture('postcss')
    expect(result.failed).toBe(false)
    expect(result.stdout).toMatchSnapshot()
  })
})
