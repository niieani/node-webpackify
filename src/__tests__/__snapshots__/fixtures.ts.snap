// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uses a loader css-loader and style-loader 1`] = `
"<head><meta charset=\\"utf-8\\"><style type=\\"text/css\\">body {
    color: black;
}
</style></head><body></body>"
`;

exports[`uses a loader postcss-loader 1`] = `
"<head><meta charset=\\"utf-8\\"><style type=\\"text/css\\">.hMP53iZXymzbgPMrMPcB8 {
    color: blanchedalmond;
    -webkit-animation: linear;
            animation: linear;
}

body {
    color: black;
}
</style></head><body></body>"
`;

exports[`uses a loader raw-loader 1`] = `"I'm a text file"`;
